gen/mailman_cesnet_user_mail

```diff
@@ -0,0 +1,110 @@
+
+my $mailinglistStruc = {};  # $mailinglistStruc->{"mailing list name including lang"}->{"user's e-mail"}->{A_USER_*}
+
+my $mailinglistsDirectory = "$DIRECTORY/mailinglists";
+mkdir $mailinglistsDirectory or die "Can't mkdir $mailinglistsDirectory: $!";
+
+#get facilityId from data
+my $facilityId = $data->getFacilityId;
+
+foreach my $resourceId ( $data->getResourceIds() ) {
+	local *FILE;
+
+	my $listVariantsAttribute = $data->getResourceAttributeValue( resource => $resourceId, attrName => $A_RESOURCE_MAILING_LIST_VARIANTS );
+	my $useLangVariants = (defined($listVariantsAttribute) && ($listVariantsAttribute eq "true") );
+
+	my $nonValidUsersAttribute = $data->getResourceAttributeValue( resource => $resourceId, attrName => $A_RESOURCE_MAILING_LIST_NONVALID_USERS );
+	my $allowNonvalidUsers = (defined($nonValidUsersAttribute) && ($nonValidUsersAttribute eq "true") );
+
+	my $disableOptOutAttribute = $data->getResourceAttributeValue( resource => $resourceId, attrName => $A_RESOURCE_DISABLE_OPT_OUT );
+	my $disableOptOut = (defined($disableOptOutAttribute) && ($disableOptOutAttribute eq "true") );
+
+	my $listNameAttribute = $data->getResourceAttributeValue( resource => $resourceId, attrName => $A_RESOURCE_MAILING_LIST_NAME );
+	my $listName = "$listNameAttribute";
+	my $listNameEN = $listName . "-en";
+	my $listNameCS = $listName . "-cs";
+
+	#create not existing mailing list in structure (even if is empty)
+	if($useLangVariants) {
+		$mailinglistStruc->{$listNameCS} = {} unless $mailinglistStruc->{$listNameCS};
+		$mailinglistStruc->{$listNameEN} = {} unless $mailinglistStruc->{$listNameEN};
+	} else {
+		$mailinglistStruc->{$listName} = {} unless $mailinglistStruc->{$listName};
+	}
+
+	foreach my $memberId ($data->getMemberIdsForResource( resource => $resourceId )) {
+
+		#list only VALID members except if allowNonvalidUsers is true
+		my $memberStatus = $data->getMemberAttributeValue( member => $memberId, attrName => $A_USER_STATUS );
+		next unless(($memberStatus eq "VALID") || ($allowNonvalidUsers));
+
+		# check if mailing list respect opt-out
+		unless ($disableOptOut) {
+			# list only members that did not opt out
+			my $userOptOutAttribute = $data->getMemberResourceAttributeValue( resource => $resourceId, member => $memberId, attrName => $A_USER_OPT_OUT );
+			next if defined $userOptOutAttribute;
+		}
+
+		my $listNameWithLang = $listName;
+		if($useLangVariants) {
+			my $userLangAttribute = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LANG );
+			if(defined($userLangAttribute) && $userLangAttribute eq "cs") {
+				$listNameWithLang = $listNameCS;
+			} else {
+				$listNameWithLang = $listNameEN;
+			}
+		}
+
+		my $userMailAttribute = $data->getMemberAttributeValue( member => $memberId, attrName => $A_USER_MAIL );
```

Comment:
You need to use "getUserAttributeValue" here. Parameters are the same.

Reply:
Fixed.